src/app/confirmation/[confirmationKey]/page.js

```diff
@@ -0,0 +1,41 @@
+"use client";
+
+import { useEffect, useState } from 'react';
+import { useRouter, useParams } from "next/navigation";
+import { useMutation } from "react-query";
+import { Error } from "@/app/activities/[activityId]/edit/components/formikFields";
+import { Container, CircularProgress } from '@mui/material';
+import { emailConfirmation } from "@/app/api.mjs";
+
+export default function EmailConfirmation({params}) {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+  const mutation = useMutation(emailConfirmation);
+  const { confirmationKey } = params;
+  const decodedKey = decodeURIComponent(confirmationKey);
+
+  useEffect(() => {
+    const confirmEmail = () => {
```

Comment:
It is hard to read, as everything is in useEffect.
To make this simpler extract a function and just call it from a hook.
Defining functions inside methods or hooks is generally not the best idea.